python/ray/serve/handle.py

```diff
@@ -1,9 +1,37 @@
-from typing import Optional, Dict, Any, Union
+import asyncio
+import threading
+from typing import Coroutine, Optional, Dict, Any, Union
+from contextlib import contextmanager
 
 from ray.serve.context import TaskContext
-from ray.serve.router import RequestMetadata
+from ray.serve.router import RequestMetadata, Router
 from ray.serve.utils import get_random_letters
 
+global_async_loop = None
+
+
+@contextmanager
```

Comment:
I wasn't familiar with this, maybe leave a comment saying that this lets you easily do `with` statements?

Reply:
also comment that it returns it to the previous state